From 1199a241df07f7162fbf06ed9d464d2b8df7f2c2 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Thu, 15 Nov 2012 17:49:30 +0800
Subject: [PATCH 220/259] ENGR00233797 Enlarge BIT_WORK_SIZE

- Enlarge BIT_WORK_SIZE to adapt to FW v2.1.8 change
- Enable dprintf on Android
- Rectify address of BIT_RESET_CTRL on i.MX6

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_debug.h | 5 +++++
 vpu/vpu_lib.c   | 2 ++
 vpu/vpu_lib.h   | 2 +-
 vpu/vpu_reg.h   | 4 ++++
 4 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/vpu/vpu_debug.h b/vpu/vpu_debug.h
index 647b8d0..55461b1 100644
--- a/vpu/vpu_debug.h
+++ b/vpu/vpu_debug.h
@@ -30,8 +30,13 @@
 
 extern int vpu_lib_dbg_level;
 
+#ifdef BUILD_FOR_ANDROID
+#define dprintf(level, fmt, arg...)     if (vpu_lib_dbg_level >= level) \
+        LOGD(fmt, ## arg)
+#else
 #define dprintf(level, fmt, arg...)     if (vpu_lib_dbg_level >= level) \
         printf("[DEBUG]\t%s:%d " fmt, __FILE__, __LINE__, ## arg)
+#endif
 
 #ifdef BUILD_FOR_ANDROID
 #ifndef ALOGE
diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index c4b9ecb..bbb1e20 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2454,6 +2454,8 @@ RetCode vpu_DecOpen(DecHandle * pHandle, DecOpenParam * pop)
 
 	/* Allocate context buffer */
 	pCodecInst->contextBufMem.size = SIZE_CONTEXT_BUF;
+	if (cpu_is_mx6x() && pop->bitstreamFormat == STD_AVC)
+		pCodecInst->contextBufMem.size += PS_SAVE_SIZE;
 	ret = IOGetPhyMem(&pCodecInst->contextBufMem);
 	if (ret) {
 		err_msg("Unable to obtain physical mem\n");
diff --git a/vpu/vpu_lib.h b/vpu/vpu_lib.h
index a992af8..fcab7f9 100644
--- a/vpu/vpu_lib.h
+++ b/vpu/vpu_lib.h
@@ -667,7 +667,7 @@ typedef struct vpu_versioninfo {
  * v4.2.2 [2008.09.03] support encoder on MX51
  * v4.0.2 [2008.08.21] add the IOClkGateSet() for power saving.
  */
-#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 7)
+#define VPU_LIB_VERSION_CODE	VPU_LIB_VERSION(5, 4, 8)
 
 extern unsigned int system_rev;
 
diff --git a/vpu/vpu_reg.h b/vpu/vpu_reg.h
index 4cdeb6c..065b34c 100644
--- a/vpu/vpu_reg.h
+++ b/vpu/vpu_reg.h
@@ -51,7 +51,11 @@
 #define CMD_DEC_DISPLAY_REORDER		0x114
 #define BIT_BIT_STREAM_PARAM		0x114
 #define BIT_TEMP_BUF_ADDR		0x118
+#if defined(IMX6Q)
+#define	BIT_RESET_CTRL			0x128
+#else
 #define	BIT_RESET_CTRL			0x11C
+#endif
 
 #define BIT_RD_PTR			0x120
 #define BIT_WR_PTR			0x124
-- 
1.8.0

